File: Restaurant-information-system/src/main/java/com/galaxy/Restaurantinformationsystem/Repository/UserRepository.java
package com.galaxy.Restaurantinformationsystem.Repository;

import com.galaxy.Restaurantinformationsystem.Entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long> {
    @Query("SELECT u FROM UserEntity u WHERE u.ID = ?1 and u.password = ?2 ")
    UserEntity readbylogin(String userID, String userPW);
}

File: src/main/java/com/galaxy/restaurant/information/system/user/repository/UserRepository.kt
package com.galaxy.restaurant.information.system.user.repository

import com.galaxy.restaurant.information.system.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    /**
     * 이메일로 사용자를 찾습니다.
     *
     * @param email 사용자 이메일
     * @return 사용자 엔티티
     */
    fun findByEmail(email: String): User?

    /**
     * 이메일로 사용자가 존재하는지 확인합니다.
     *
     * @param email 사용자 이메일
     * @return 사용자가 존재하면 true, 존재하지 않으면 false
     */
    fun existsByEmail(email: String): Boolean
}
